libraries/type-length-value/js/src/splDiscriminate.ts

```diff
@@ -0,0 +1,17 @@
+import type { TlvDiscriminator } from './tlvState.js';
+import { createHash } from 'crypto';
+
+export class SplDiscriminator implements TlvDiscriminator {
+    private readonly discriminator: string;
+    private readonly length: number;
+
+    public constructor(discriminator: string, length = 8) {
+        this.discriminator = discriminator;
+        this.length = length;
+    }
+
+    public get bytes(): Buffer {
+        const digest = createHash('sha256').update(this.discriminator).digest();
+        return digest.subarray(0, this.length);
+    }
+}
```

Comment:
This looks good, but can you run a manual test to make sure this function creates the same hash as the `solana_program` hash? Yes, they're both sha256 but want to be absolutely sure.

You don't have to commit anything for that though.

Reply:
Can add a simple test case with static bytes!